### PR TITLE
Only run codegen at start of flutter_test

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Only run codegen once at the start of flutter test. allow initializing the codegen compiler to skip subsequent code generation and only configure to accept codegen. Add test to bots.dart that executes a test which requires code generation.

## Related Issues

https://github.com/flutter/flutter/issues/27141

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes tests for *all* changed/updated/fixed behaviors (See [Test Coverage]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/test.dart [hunk 1]
```diff
@@ -159,6 +160,17 @@ class TestCommand extends FlutterCommand {
 
     Cache.releaseLockEarly();
 
+    // Run builders once before all tests.
+    if (experimentalBuildEnabled && await flutterProject.hasBuilders) {
+      final CodegenDaemon codegenDaemon = await codeGenerator.daemon(flutterProject);
+      codegenDaemon.startBuild();
+      await for (CodegenStatus status in codegenDaemon.buildResults) {
+        if (status == CodegenStatus.Succeeded || status == CodegenStatus.Failed) {
```
Comment (yjbanov) [inline packages/flutter_tools/lib/src/commands/test.dart]: Do we not want to issue a warning (like above) if the status is "Failed"?

**Review by yjbanov — APPROVED** [review]

File: packages/flutter_tools/lib/src/codegen.dart [hunk 2]
```diff
@@ -191,6 +188,17 @@ class CodeGeneratingResidentCompiler implements ResidentCompiler {
       // Pass an invalid file name to prevent frontend_server from initializing from dill.
       initializeFromDill: 'none_file',
     );
+    if (runCold) {
+      return residentCompiler;
+    }
+    final CodegenDaemon codegenDaemon = await codeGenerator.daemon(flutterProject);
+    codegenDaemon.startBuild();
+    final CodegenStatus status = await codegenDaemon.buildResults.firstWhere((CodegenStatus status) {
```
[the remark this answers is not in the document]
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/codegen.dart]: It generally throws a state error if it cannot initialize.

File: packages/flutter_tools/lib/src/commands/test.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/commands/test.dart]: Done

**Comment by yjbanov** [comment]
lgtm

**Pull request merged**